packages/aws-cdk/lib/util/asset-publishing.ts

```diff
@@ -215,4 +217,48 @@ class PublishingProgressListener implements cdk_assets.IPublishProgressListener
     const handler = this.quiet && type !== 'fail' ? debug : EVENT_TO_LOGGER[type];
     handler(`[${event.percentComplete}%] ${type}: ${event.message}`);
   }
+}
+
+const PUBLISH_ASSET_PROOF_SYM = Symbol('publish_assets_proof');
+
+/**
+ * This interface represents proof that assets have been published.
+ *
+ * Objects of this type can only be obtained by calling functions in this module.
+ */
+export interface AssetsPublishedProof {
+  [PUBLISH_ASSET_PROOF_SYM]: true;
+}
```

Comment:
Using `Symbol` vs `Symbol.for()` will prevent anyone else from implementing `AssetsPublishedProof` - which might well be what you want. But than why make it an interface and/or export it. This feels like a recipe for confusion in the future.

Reply:
This is exactly what I want -- I want any bit of code (also in other modules) be able to *require* an `AssetsPublishedProof`, while only this module can *generate* an instance of one.